### PR TITLE
Adds Authenticated WebViews

### PR DESCRIPTION
**Pull request by redreceipt**
### What does this PR do, or why is it needed?

Gives us the ability to query for cookies making our webviews function as if I was logged in.

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [#495]
- [x] Adds Rock Cookie context
- [x] Links `react-native-webiew`
- [x] Adds a cookie-enabled component
- [ ] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android
- [ ] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by redreceipt** [comment]
@vinnyjth I'll probably need a 30,000 ft view chat with this one at some point

**Comment by vinnyjth** [comment]
@redreceipt when you are done making changes to this PR (you might be already...) mark as "ready-for-review" and assign me and I will review it :)

**Comment by redreceipt** [comment]
> @redreceipt when you are done making changes to this PR (you might be already...) mark as "ready-for-review" and assign me and I will review it :)

not quite ready yet, hoping you could give me another high-level overview of what we're going for here at some point

**Comment by vinnyjth** [comment]
>not quite ready yet, hoping you could give me another high-level overview of what we're going for here at some point

Thanks for asking, my bad for not providing that clarity in the original issue. Let's sync up on a call tomorrow and cover the high level objectives, although from what I can see so far of this PR I think you're doing all the right things.

**Comment by redreceipt** [comment]
@vinnyjth since you questioned a couple of your design choices, do you think it would be valuable to have someone else review this? Or do you have a good idea of what to fix?

**Comment by vinnyjth** [comment]
@redreceipt I will do a review of this PR and leave comments on stuff that I think should change, after that, you can assign me to review along with someone else and go from there.

File: packages/apolloschurchapp/src/tabs/connect/ActionTable/index.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ import {
 } from '@apollosproject/ui-kit';
 import { WebBrowserConsumer } from 'apolloschurchapp/src/ui/WebBrowser';
 import NavigationActions from 'apolloschurchapp/src/NavigationService';
+// import { UserWebBrowserConsumer } from 'apolloschurchapp/src/user-web-view';
```
Comment (vinnyjth) [inline packages/apolloschurchapp/src/tabs/connect/ActionTable/index.js]: We can remove this :)

File: packages/apollos-church-api/src/data/authenticated-user/index.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import gql from 'graphql-tag';
```
Comment (vinnyjth) [inline packages/apollos-church-api/src/data/authenticated-user/index.js]: So sorry - something I missed from the first time around. In the christ fellowship code, we had to create a new resolver and extend the schema because we didn't have access to the original schema. In this case, instead of editing and overriding an existing type, we should change the original type / resolver. You will find the schema in the `data-schema` package, and the `AuthenticatedUser` is in `src/auth/index.js`

**Comment by vinnyjth** [comment]
@redreceipt There's one more structural change that I didn't notice earlier that I noted above :) 

Beyond that, this is looking real good! I'll review it on the device when I get a chance.

**Comment by redreceipt** [comment]
@vinnyjth which resolver? `apollos-data-connector-rock-auth/lib` or `apollos-data-connector-rock/src/auth/`?

**Comment by vinnyjth** [comment]
@redreceipt `apollos-data-connector-rock-auth` shouldn't exist any more, so the latter.

**Comment by redreceipt** [comment]
@vinnyjth not sure how to proceed on the codecov fail. Don't even really know what codecov is...

**Comment by vinnyjth** [comment]
@redreceipt Codecov is a tool we use to keep track of how much our codebase is "covered" by tests. Adding coverage over this filed is probably pretty easy :) let's pair on it later today.

**Comment by vinnyjth** [comment]
👀 ing right now

**Comment by vinnyjth** [comment]
Looking great @redreceipt :D I added a way to test the webview, and then needed to tweak some ways the navigation props was being handled, but now we are good.
